forms/core/Application.php

```diff
@@ -1468,4 +1398,34 @@ public static function encode($string){
         return stripslashes(htmlspecialchars($string));
     }
 
+    public static function refreshTokenIfNeeded() {
+		$auth = array(
+			'access_token' => Settings::get('access_token'),
+			'refresh_token' => Settings::get('refresh_token')
+		);
+
+		if (!empty($auth['access_token']) && Settings::get('expiry') < time()) {
+            $newCredentials = Application::$CampaignMonitor->refresh_token($auth);
+			
+			if (isset($newCredentials->error)) {
+				// TODO: add some kind of error-handling if API returns error
+				// For now just clear out the tokens
+				Application::updateTokens('', '', 0);

```

Comment:
Are we sure that existing data are definitely not needed anymore? If it's a temporary network issue, we may still want to refresh the token in the future but potentially that information might be wiped away here?